src/ja/server/diagrams/schedulerLowLevel.puml

```diff
@@ -0,0 +1,129 @@
+@startuml
+
+box "Central Server" #Lightblue
+    participant ":QueuingActivity"
+    participant ":Scheduler"
+    participant "job:Job"
+    participant ":RunningJobEntry"
+    participant ":DataBaseHandler"
+    participant ":Dispatcher"
+end box
+
+activate "job:Job"
+-> ":QueuingActivity":
+activate ":QueuingActivity"
+
+":QueuingActivity" -> ":Scheduler" ++: queueJob(job:Job)
+
+note right of ":QueuingActivity"
+  user calls 
+  "add" command
+end note
+
+":Scheduler" -> "job:Job" ++: getPriority(job)
+"job:Job" --> ":Scheduler" --: p: Priority
+
+":Scheduler" -> ":Scheduler" ++: findWC(job)
+":Scheduler" -> ":DataBaseHandler" ++: getWCList()
+note left of ":Scheduler"
+    Scheduler decides which jobs
+    should run on each machine
+end note
+":DataBaseHandler" --> ":Scheduler" --: WCList:list[]
+":Scheduler" --> ":Scheduler" --: wmID:int
+
+":Scheduler" -> "job:Job" ++: set_wm_ID(wmID)
+"job:Job" --> ":Scheduler" --:
+
+":Scheduler" -> ":RunningJobEntry" ++: <<create>> 
+note over ":RunningJobEntry" #ba69bf
+    attributes to be intialized
+    *jobID
+    *wmID
+end note    
+":RunningJobEntry" --> ":Scheduler" --: rje
+break   p==URGENT
+    ":Scheduler" -> ":DataBaseHandler" ++: addRunningJob(rje)
+    note right of ":DataBaseHandler" 
+        RunningJobEntry contains the
+        job ID and work machine ID
+    end note
+    ":DataBaseHandler" --> ":Scheduler" --:
+    ":Scheduler" -> ":Dispatcher" ++: dispatchUrgentJob(job)
+    ":Dispatcher" -> ":Dispatcher" ++: releaseResources(wmID)
+    return pausedJobs:list[RunningJobEntry]
+
+
+    note right of ":Dispatcher" 
+        if enough resources
+        are present no jobs
+        will be paused        
+    end note
+
+    loop rj in pausedJobs
+    ":Dispatcher" -> ":Dispatcher" ++: pause(rj)
+    ":Dispatcher" --> ":Dispatcher" --: 
+    end loop
+
+    ":Dispatcher" -> ":Dispatcher" ++: start(job)
+    ":Dispatcher" --> ":Dispatcher" --: message
+
+    ":Dispatcher" --> ":Scheduler": message
+    ":Scheduler" --> ":QueuingActivity": message
+    <-- ":QueuingActivity" : message
+
+    ... ...
+    loop rj in pausedJobs
+        ":Dispatcher" -> ":Dispatcher" ++: resume(rj)
+        ":Dispatcher" --> ":Dispatcher" --: 
+    end loop
+    deactivate ":Dispatcher"
+
+end
+
+":Scheduler" -> ":DataBaseHandler" ++: checkIdle(wmID)
+note right of ":DataBaseHandler"
+    check if there are enough 
+    available hardware resources
+end note
+":DataBaseHandler" --> ":Scheduler" --: isIdle:bool
+
+break isIdle
+    ":Scheduler" -> ":DataBaseHandler" ++: addRunningJob(rje)
+    ":DataBaseHandler" --> ":Scheduler" --:
+
+    ":Scheduler" -> ":Dispatcher" ++: dispatchJob(job)
+    ":Dispatcher" -> ":Dispatcher" ++: start(job)
+    ":Dispatcher" --> ":Dispatcher" --: message
+    ":Dispatcher" --> ":Scheduler": message
+    deactivate ":Dispatcher"
+    ":Scheduler" --> ":QueuingActivity": message
+    <-- ":QueuingActivity" : message
+
+end
+
+":Scheduler" -> "job:Job" ++: calcCost()
```

Comment:
Job cost functions must be evaluated **before** the Scheduler runs, not afterwards.